Dockerfile

```diff
@@ -1,32 +1,68 @@
 # DockerFile for grpc4bmi. Installs the C++ bindings in the /usr/local prefix directory in the container. If you are
 # planning to run a container with your BMI-enabled model and communicate with it using grpc4bmi, you can use this as a
 # base image for your model
-FROM ubuntu:bionic
-MAINTAINER Gijs van den Oord <g.vandenoord@esciencecenter.nl>
-RUN apt-get update
+FROM ubuntu:24.04
+LABEL maintainer="eWaterCycle <ewatercycle@esciencecenter.nl>"
+LABEL org.opencontainers.image.source="https://github.com/eWaterCycle/grpc4bmi"
+
+ENV GRPC_VERSION="1.66.1"
+ENV BMIC_VERSION="2.1.2"
+ENV BMICXX_VERSION="2.0.2"
 
 # Prerequisite packages
-RUN apt-get install -y wget git build-essential g++ make cmake curl automake libtool pkg-config gfortran
+RUN apt-get update && apt-get install -y \
+    automake \
+    build-essential \
+    cmake \
+    curl \
+    g++ \
+    gfortran \
+    git \
+    libtool \
+    libssl-dev \
+    make \
+    pkg-config \
+    vim-tiny \
+    wget \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
 
 # Build grpc from source
-RUN git clone -b $(curl -L https://grpc.io/release) --depth=1 https://github.com/grpc/grpc /opt/grpc
+RUN git clone -b v${GRPC_VERSION} --depth 1 https://github.com/grpc/grpc /opt/grpc
 WORKDIR /opt/grpc
 RUN git submodule update --init --recursive
-RUN make install
-WORKDIR third_party/protobuf
-RUN make install
+WORKDIR /opt/grpc/cmake/_build
+RUN cmake ../.. \
+        -DgRPC_INSTALL=ON \
+        -DgRPC_SSL_PROVIDER=package \
+        -DgRPC_BUILD_TESTS=OFF \
+        -DCMAKE_CXX_STANDARD=17 \
+        -DBUILD_SHARED_LIBS=ON && \
+    make install && \
+    make clean
 
-# Build bmi-c from source
-RUN git clone --depth=1 https://github.com/eWaterCycle/grpc4bmi.git /opt/grpc4bmi
-WORKDIR /opt/grpc4bmi
-RUN git submodule update --init --recursive
-RUN mkdir -p /opt/grpc4bmi/cpp/bmi-c/build
-WORKDIR /opt/grpc4bmi/cpp/bmi-c/build
-RUN cmake ..
-RUN make install
+# Build bmi-c and bmi-cxx from source
+RUN git clone --branch v${BMIC_VERSION} https://github.com/csdms/bmi-c /opt/bmi-c
+WORKDIR /opt/bmi-c/_build
+RUN cmake .. && \
+    make install && \
+    make clean
+RUN git clone --branch v${BMICXX_VERSION} https://github.com/csdms/bmi-cxx /opt/bmi-cxx
+WORKDIR /opt/bmi-cxx/_build
+RUN cmake .. && \
+    make install && \
+    make clean
+
+RUN ldconfig
 
 # Build grpc4bmi from source
-RUN mkdir -p /opt/grpc4bmi/cpp/build
-WORKDIR /opt/grpc4bmi/cpp/build
-RUN cmake ..
-RUN make install
+RUN git clone --branch update-dockerfile --depth 1 https://github.com/csdms/grpc4bmi /opt/grpc4bmi
+WORKDIR /opt/grpc4bmi
+RUN git submodule update --init
+WORKDIR /opt/grpc4bmi/cpp/_build
+RUN cmake .. -DCMAKE_CXX_STANDARD=17 && \
```

Comment:
A possible improvement could be to isolate the build env from the final container. This would make the final docker image size a lot smaller. For an example, see https://github.com/haddocking/powerfit/blob/96d0e7519ec1610ec951119dc93c3d98cdc3b3a3/Dockerfile

However, I guess a lot of the build dependencies are also required again for building the model containers, so maybe it's better to do that there :thinking: